### DISCUSSION TITLE
Type 0 Resource Allocation

### DISCUSSION THREAD

**Discussion by arman-maghsoudnia** [srsran/srsRAN_Project#331] [General Help] [answered]
Dear Community,

From what I understand, reading the code, it seems that srsRAN only does Type 1 resource allocation. Is Type 0 resource allocation not supported yet?
I have another related question as well. It seems that srsRAN only support one Bandwidth Part (BWP), which is always the same, for all UEs. And this BWP covers the whole available Bandwidth. Only during inactivity time, the UEs will switch to a smaller BWP which I guess would be the default BWP. Is that really the case?

**Comment by herlesupreeth** [comment, marked as answer]
> From what I understand, reading the code, it seems that srsRAN only does Type 1 resource allocation. Is Type 0 resource allocation not supported yet?

That is correct.

> I have another related question as well. It seems that srsRAN only support one Bandwidth Part (BWP), which is always the same, for all UEs. And this BWP covers the whole available Bandwidth. Only during inactivity time, the UEs will switch to a smaller BWP which I guess would be the default BWP. Is that really the case?

Yes, srsRAN implements only one BWP and during inactivity time as well its the same BWP but the nof. RBs that can be used are limited to CORESET#0 RBs as mentioned in the specification.

**Reply by improperaffo** [reply]
Is there a plan to implement Type 0 resource allocation in the coming releases?

**Reply by herlesupreeth** [reply]
This feature is not planned to be implemented yet.
